lib/cDebug.ml

```diff
@@ -0,0 +1,45 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+let debug : (string, int) Hashtbl.t = Hashtbl.create 97
```

Comment:
It's unclear to me why this is not a `Summary.ref` given that one can modify it by `Set Debug...` and then it must be a functional map)

Reply:
The goptions system (which is used to provide Set Debug) assumes regular mutation, not up to summary mutation.
This allows using it for lots of things including hashtables in cwarnings and now cdebug.